paper.md

```diff
@@ -0,0 +1,160 @@
+multi-language projects Julia often acts as a glue code between newly developed
+implementations written in Julia and existing libraries written in C/Fortran.
+
+While there exist other numerical simulation codes in Julia, e.g., Gridap.jl [@Badia2020] or
+Ferrite.jl [@Carlsson_Ferrite_jl], none of them provide the ability to use them directly
+from another programming language. With `libtrixi`, we therefore enable new workflows by
+allowing scientists to connect established research codes to a modern numerical simulation
+package written in Julia. That is, a main program written in C/C++ or Fortran is able to
+execute a simulation set up in Julia with Trixi.jl without sacrifices in performance.
+
+This control direction, where a Julia library is managed from C/Fortran, has been rarely
+used in practice so far. Besides making Trixi.jl available to a wider scientific audience,
+`libtrixi` is thus a research project to investigate the efficacy of using Julia-based
+libraries in existing code environments, and can eventually serve as a blueprint for
+other similar efforts. Questions such as how to retain the flexibility of Julia while
+providing a traditional, fixed API, how to use system-local third-party libraries, or how to
+interact with Fortran are investigated and answered.
+
+
+# Technical overview
+`Libtrixi` consists of three main parts:
+
+- the Julia package
+  [LibTrixi.jl](https://github.com/trixi-framework/libtrixi/tree/main/LibTrixi.jl), which
+  provides a traditional library API to Trixi.jl in Julia,
+- a C API that exposes this Julia API as an ordinary, shared C library,
+- Fortran bindings for the C API.
+
+\autoref{fig:libtrixi_overview} illustrates the general workflow: A main program written in
+C/C++/Fortran interacts with the public-facing C or Fortran API of `libtrixi`. The
+Fortran API is little more than a set of language bindings, with some extra code to handle
+the conversion between Fortran strings and C strings. The C API is slightly more involved,
+providing functionality to initialize and eventually finalize the Julia runtime. During
+initialization, C function pointers are obtained from Julia to the relevant API functions
+implemented in LibTrixi.jl. Most of the C API then just forwards the API calls to these
+function pointers.
+
+Finally, LibTrixi.jl is the actual library layer. Since Trixi.jl uses a composable design
+and relies heavily on Julia's type system, it is necessary to repackage this flexibility
+such that it can be exposed in a traditional API with static types. This is achieved by
+converting Trixi.jl's *elixirs*, which consist of all code to set up and run a simulation,
+into so-called *libelixirs*. These libelixirs are used to intialize a simulation state,
+which is then stored internally in LibTrixi.jl and assigned a unique integer. This integer
+handle is exposed in the C/Fortran API and facilitates all interaction between the main
+program and the actual simulation.
```

Comment:
It also allows to control multiple simulations concurrently (which however is not tested and therefore should probably not be mentioned?)